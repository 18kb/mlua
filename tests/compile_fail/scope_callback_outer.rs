use mlua::{Lua, Table, Result};

struct Test {
    field: i32,
}

fn main() {
    let lua = Lua::new();
    let mut outer: Option<Table> = None;
    lua.scope(|scope| -> Result<()> {
        let f = scope
            .create_function_mut(|_, t: Table| {
                outer = Some(t);
                Ok(())
            })?;
        f.call::<_, ()>(lua.create_table()?)?;
        Ok(())
    });
}
