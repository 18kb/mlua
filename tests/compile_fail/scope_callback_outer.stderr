error: borrowed data cannot be stored outside of its closure
  --> $DIR/scope_callback_outer.rs:11:17
   |
9  |     let mut outer: Option<Table> = None;
   |         --------- ...so that variable is valid at time of its declaration
10 |     lua.scope(|scope| -> Result<()> {
   |               --------------------- borrowed data cannot outlive this closure
11 |         let f = scope
   |                 ^^^^^ cannot be stored outside of its closure
12 |             .create_function_mut(|_, t: Table| {
   |              ------------------- cannot infer an appropriate lifetime...
